src/lib/rest/RestService.cpp

```diff
@@ -352,6 +352,90 @@ static void filterRelease(ParseData* parseDataP, RequestType request)
 
 /* ****************************************************************************
 *
+* compCheck - 
+*/
+static bool compCheck(int components, const std::vector<std::string>& compV)
+{
+  for (int ix = 0; ix < components; ++ix)
+  {
+    if (compV[ix] == "")
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+
+/* ****************************************************************************
+*
+* compErrorDetect - 
+*/
```

Comment:
A bit cryptic... maybe comments over each if branch in the logic would help to understand how it works (making future maintenance easier).